src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -29,34 +31,75 @@
      */
     public EditCommand parse(String args) throws ParseException {
         requireNonNull(args);
-        ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_GENDER, PREFIX_TAG);
+        PersonType personType = ParserUtil.parsePersonType(args);
+        switch (personType) {
+        case TUTOR:
+            ArgumentMultimap tutorArgMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_GENDER,
+                    PREFIX_QUALIFICATION, PREFIX_TAG);
 
-        Index index;
+            Index tutorIndex;
 
-        try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
-        }
+            try {
+                tutorIndex = ParserUtil.parseIndex(tutorArgMultimap.getPreamble());
+            } catch (ParseException pe) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
+            }
 
-        EditPersonDescriptor editPersonDescriptor = new EditPersonDescriptor();
-        if (argMultimap.getValue(PREFIX_GENDER).isPresent()) {
-            editPersonDescriptor.setGender(ParserUtil.parseGender(argMultimap.getValue(PREFIX_GENDER).get()));
-        }
-        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
-            editPersonDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()));
-        }
-        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
-            editPersonDescriptor.setPhone(ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get()));
-        }
-        parseTagsForEdit(argMultimap.getAllValues(PREFIX_TAG)).ifPresent(editPersonDescriptor::setTags);
+            EditTutorDescriptor editTutorDescriptor = new EditTutorDescriptor();
+            if (tutorArgMultimap.getValue(PREFIX_NAME).isPresent()) {
+                editTutorDescriptor.setName(ParserUtil.parseName(tutorArgMultimap.getValue(PREFIX_NAME).get()));
+            }
+            if (tutorArgMultimap.getValue(PREFIX_PHONE).isPresent()) {
+                editTutorDescriptor.setPhone(ParserUtil.parsePhone(tutorArgMultimap.getValue(PREFIX_PHONE).get()));
+            }
+            if (tutorArgMultimap.getValue(PREFIX_GENDER).isPresent()) {
+                editTutorDescriptor.setGender(ParserUtil.parseGender(tutorArgMultimap.getValue(PREFIX_GENDER).get()));
+            }
+            if (tutorArgMultimap.getValue(PREFIX_QUALIFICATION).isPresent()) {
+                editTutorDescriptor.setQualification(ParserUtil.parseQualification(
+                        tutorArgMultimap.getValue(PREFIX_QUALIFICATION).get()));
+            }
+            parseTagsForEdit(tutorArgMultimap.getAllValues(PREFIX_TAG)).ifPresent(editTutorDescriptor::setTags);
 
```

Comment:
I think it's possible to extract this out as well. We can further discuss this later on!